build-system/babel-plugins/babel-plugin-transform-rename-privates/index.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Renames all of AMP's class properties to have a unique suffix, `AMP_PRIVATE_`.
+ * This acts as an indicator for terser to mangle the names even when combined with 3p files.
+ *
+ * @param {babel} babel
+ * @interface {babel.PluginPass}
+ * @return {babel.PluginObj}
+ */
+module.exports = function (babel) {
+  const {types: t} = babel;
+
+  /**
+   * Adds trailing AMP_PRIVATE_ suffix to private identifiers.
+   * @param {string} field
+   * @return {function(*,*):void}
+   */
+  function renamePrivate(field) {
+    return function (path, state) {
+      if (!isAmpSrc(state)) {
+        return;
+      }
+      if (path.node.computed) {
+        return;
+      }
+
+      const key = path.get(field);
+      if (!key.isIdentifier()) {
+        return;
+      }
+
+      const {name} = key.node;
+      if (name.endsWith('_AMP_PRIVATE_')) {
+        return;
+      }
+
+      if (!name.endsWith('_')) {
+        return;
+      }
+      key.replaceWith(t.identifier(`${name}AMP_PRIVATE_`));
+    };
+  }
+
+  return {
+    visitor: {
+      Method: renamePrivate('key'),
+      Property: renamePrivate('key'),
+      MemberExpression: renamePrivate('property'),
+      OptionalMemberExpression: renamePrivate('property'),
+    },
+  };
+};
+
+/**
+ * @param {*} state
+ * @return {string}
+ */
+function isAmpSrc(state) {
+  const filename = state.file.opts.filenameRelative;
+  if (!filename) {
+    throw new Error('Cannot use plugin without providing a filename');
+  }
+  return filename.startsWith('src/') || filename.startsWith('extensions/');
```

Comment:
I think we should invert this, and check for `third_party` and `node_modules`, to match the Closure behavior.

Reply:
I didn't know closure did that. updating!

Reply:
Done

Reply:
Unit tests are breaking when I do this, but I bet its a random conflict unrelated to this PR. I'm investigating now.

Reply:
Don't have time to figure this out before weekend plans. Kicking it to next week

Reply:
`amp-render` tests always fail if they happen after `amp-iframe`